src/main/scala/ai/privado/passes/DBTParserPass.scala

```diff
@@ -0,0 +1,462 @@
+                  dbKeys.foreach { k =>
+                    if (outputData.containsKey(k)) {
+                      result = (
+                        profileFile,
+                        k,
+                        outputData.get(k).asInstanceOf[String],
+                        outputData.getOrDefault("host", "").asInstanceOf[String],
+                        outputData.getOrDefault("type", "").asInstanceOf[String]
+                      )
+                      break
+                    }
+                  }
+                }
+
+              case _ =>
+                logger.error(f"could not find any output data in profiles: ${profileFile}")
+            }
+          }
+        }
+      }
+    }
+    result
+  }
+
+  private def getProfileOutputData(data: java.util.Map[String, Any], defaultOutput: String) = {
+    // look in production outputs
+    var result: Option[(String, java.util.Map[String, Any])] = None
+
+    val productionKeys = Array("production", "prod", "prd")
+    productionKeys.foreach(x => {
+      if (result.isEmpty && data.containsKey(x)) {
+        result = Some(x, data.get(x).asInstanceOf[java.util.Map[String, Any]])
+      }
+    })
+
+    // look in default output
+    if (result.isEmpty && data.containsKey(defaultOutput)) {
+      result = Some(defaultOutput, data.get(defaultOutput).asInstanceOf[java.util.Map[String, Any]])
+    }
+
+    // look in first (or any)
+    if (result.isEmpty && !data.isEmpty) {
+      val headKeyValue = data.entrySet().iterator().next()
+      result = Some(headKeyValue.getKey, headKeyValue.getValue.asInstanceOf[java.util.Map[String, Any]])
+    }
+
+    result
+  }
+  private def getModelsDirectoryName(data: java.util.Map[String, Any]) = {
+    data.get("model-paths") match {
+      case value: java.util.List[_] if !value.isEmpty =>
+        value.get(0).toString
+      case value: String =>
+        value
+      case _ =>
+        "models"
+    }
+  }
+
+  private def getYAML(file: String) = Try {
+    val yamlContent = better.files.File(file).contentAsString // Read the YAML file content as a string
+    val yaml        = new Yaml(new SafeConstructor(LoaderOptions()))
+    yaml.load(yamlContent).asInstanceOf[java.util.Map[String, Any]]
+  }
+
+  private def getConfigFiles(
+    projectRoot: String,
+    extensions: Set[String],
+    allowedFiles: Set[String] = Set()
+  ): List[String] = {
+    def getListOfFiles(dir: String): List[File] = {
+      val d = new File(dir)
+      if (d.exists && d.isDirectory) {
+        d.listFiles.filter(_.isFile).toList
+      } else {
+        List[File]()
+      }
+    }
+
+    SourceFiles
+      .determine(Set(projectRoot), extensions)
+      .concat(
+        getListOfFiles(projectRoot)
+          .map(f => {
+            f.getAbsolutePath
+          })
+      )
+      .filter(_.matches(f".*(${allowedFiles.mkString("|")})"))
+      .filter(file => Utilities.isFileProcessable(file, ruleCache) && (!file.matches(".*node_modules.*")))
```

Comment:
Is this needed `(!file.matches(".*node_modules.*")))`?

Reply:
Again, this was used in the PropertyPass, assumed that we need this filter?